core/html_api.php

```diff
@@ -1390,10 +1390,22 @@ function html_status_legend( $p_display_position, $p_restrict_by_filter = false
 	$t_workflow = config_get( 'status_enum_workflow' );
 	if( !empty( $t_workflow ) ) {
 		foreach( $t_status_array as $t_status => $t_name ) {
+			# Check if the workflow allows leaving the state
 			if( !isset( $t_workflow[$t_status] ) ) {
 
+				# Check if the workflow allows entering the state
+				$t_can_enter = false;
+				foreach( $t_workflow as $t_values ) {
+					if( MantisEnum::hasValue( $t_values, $t_status ) ) {
+						$t_can_enter = true;
+						break;
+					}
+				}
+
 				# drop elements that are not in the workflow
-				unset( $t_status_array[$t_status] );
+				if( !$t_can_enter ) {
+					unset( $t_status_array[$t_status] );
```

Comment:
Out of curiosity, I setup a [quick test ](http://ideone.com/re3qeN) which demonstrates that unsetting an array element within the foreach has no impact on the loop variables themselves, although a PHP notice is triggered (as expected) when the accessing the array element that was unset. When iterating by reference, the removed value is not even processed.